addons/mail/static/src/components/discuss/tests/discuss_tests.js

```diff
@@ -3513,6 +3525,137 @@ QUnit.test('receive new needaction messages', async function (assert) {
     );
 });
 
+QUnit.test('should not display unfollow button when current user is not follower of the thread', async function (assert) {
+    assert.expect(1);
+
+    this.data['res.partner'].records.push({
+        id: 21,
+        message_follower_ids: [1],
+    });
+    this.data['mail.followers'].records.push({
+        id: 1,
+        partner_id: this.data.currentPartnerId,
+        res_id: 21,
+        res_model: 'res.partner',
+    });
+    this.data['mail.message'].records.push({
+        body: "<p>Test</p>",
+        id: 100, // random unique id, will be used to link notification to message
+        // needaction needs to be set here for message_fetch domain, because
+        // mocked models don't have computed fields
+        needaction: true,
+        model: 'res.partner',
+        record_name: 'Refactoring',
+        res_id: 20,
+    });
+    this.data['mail.notification'].records.push({
+        mail_message_id: 100, // id of related message
+        res_partner_id: this.data.currentPartnerId, // must be for current partner
+    });
```

Comment:
you can remove all of those comments, it was a mistake adding them originally (they are not really useful)